dex/elrond_dex_router/src/factory.rs

```diff
@@ -36,7 +36,7 @@ pub trait FactoryModule {
     ) -> SCResult<Address> {
         require!(self.pair_code_ready().get(), "Pair code not ready");
         let code_metadata = CodeMetadata::UPGRADEABLE;
-        let gas_left = self.blockchain().get_gas_left();
+        let gas_left = self.blockchain().get_gas_left() / 2;
```

Comment:
why do you divide by 2 ? it should work without it

Reply:
there are other operations happening after actual deploy, so a part of gas has to be reserved for them

Reply:
you will always get back the unused gas. so there is no need for / 2.

Reply:
removed / 2